plugins/ingest-attachment/src/test/java/org/elasticsearch/ingest/attachment/AttachmentProcessorTests.java

```diff
@@ -260,22 +269,218 @@ public void testNullWithoutIgnoreMissing() throws Exception {
+        assertThat(attachmentData, hasEntry("access_permission:can_modify", "true"));
+        assertThat(attachmentData, hasEntry("pdf:docinfo:producer", "LibreOffice 5.2"));
+        assertThat(attachmentData, hasEntry("pdf:docinfo:created", "2016-09-30T13:19:58Z"));
+        assertThat(attachmentData, hasEntry("Content-Type", "application/pdf"));
+    }
+
+    public void testWildcardFilteredRawMetadataFromPdf() throws Exception {
+        processor = new AttachmentProcessor(randomAsciiOfLength(10), "source_field",
+            "target_field", Collections.emptySet(), Collections.singleton("pdf:*"), 10000, false);
+        Map<String, Object> attachmentData = parseDocument("test.pdf", processor);
+
+        // We check that we have all expected field starting with "pdf:"
+        assertThat(attachmentData, hasEntry("pdf:PDFVersion", "1.4"));
+        assertThat(attachmentData, hasEntry("pdf:docinfo:creator_tool", "Writer"));
+        assertThat(attachmentData, hasEntry("pdf:encrypted", "false"));
+        assertThat(attachmentData, hasEntry("pdf:docinfo:producer", "LibreOffice 5.2"));
+        assertThat(attachmentData, hasEntry("pdf:docinfo:created", "2016-09-30T13:19:58Z"));
+
+        // We check that we did not extract any other field
+        assertThat(attachmentData, not(hasKey("X-Parsed-By")));
+    }
+
+    public void testFilteredRawMetadataFromPdf() throws Exception {
+        processor = new AttachmentProcessor(randomAsciiOfLength(10), "source_field",
+            "target_field", Collections.emptySet(), Collections.singleton("pdf:PDFVersion"), 10000, false);
+        Map<String, Object> attachmentData = parseDocument("test.pdf", processor);
+
+        // We check that we have only the expected field
+        assertThat(attachmentData, hasEntry("pdf:PDFVersion", "1.4"));
+
+        // We check that we did not extract any other field
+        assertThat(attachmentData, not(hasKey("pdf:encrypted")));
+    }
+
+    public void testRawMetadataWith2FiltersFromPdf() throws Exception {
+        processor = new AttachmentProcessor(randomAsciiOfLength(10), "source_field",
+            "target_field", Collections.emptySet(), Sets.newHashSet("pdf:PDFVersion", "pdf:encrypted"), 10000, false);
+        Map<String, Object> attachmentData = parseDocument("test.pdf", processor);
+
+        // We check that we have only expected fields
+        assertThat(attachmentData, hasEntry("pdf:PDFVersion", "1.4"));
+        assertThat(attachmentData, hasEntry("pdf:encrypted", "false"));
+
+        // We check that we did not extract any other field
+        assertThat(attachmentData, not(hasKey("pdf:docinfo:creator_tool")));
+    }
+
+    public void testWildcardAndFiltersRawMetadataFromPdf() throws Exception {
+        processor = new AttachmentProcessor(randomAsciiOfLength(10), "source_field",
+            "target_field", Collections.emptySet(), Sets.newHashSet("pdf:PDFVersion", "pdf:*"), 10000, false);
```

Comment:
you could randomize this with one of the tests above, where you randomly at an existing field to the `pdf:*` hash set?